org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.java

```diff
@@ -516,6 +516,7 @@ protected List<Diagnostic> toDiagnostics(Iterable<? extends Issue> issues) {
 	protected Diagnostic toDiagnostic(Issue issue) {
 		Diagnostic result = new Diagnostic();
 		result.setCode(issue.getCode());
+		result.setData(issue.getData());
```

Comment:
does this work on the deserialization side when the client sents back data in org.eclipse.lsp4j.CodeActionContext.getDiagnostics()

Reply:
what does lsp4j deserialize as object there

Reply:
Since it's a String[] array I expect this to become a List<String> in the Diagnostic.

Reply:
i am asking if it works somehow ootb or if we need to ship custom gson adapters for "our" diagnostics

Reply:
@cdietrich:
We tested the quickfix during development. The deserialization works via casting diagnostic.getData() to a JsonArray.
Testing confirms that a JsonArray of string literals comes out from the quickfix side.

Reply:
Thx.